Firmware code/smart_energy_meter/lib/unix_device/device_functions.c

```diff
@@ -0,0 +1,62 @@
+#include "device_functions.h"
```

Comment:
I don't think you need any of the `lib/unix_device/*` files, could you remove them?